testing/simapp/app.go

```diff
@@ -923,11 +975,22 @@ func initParamsKeeper(appCodec codec.BinaryCodec, legacyAmino *codec.LegacyAmino
 	return paramsKeeper
 }
 
+func makeEncodingConfig() simappparams.EncodingConfig {
+	encodingConfig := simappparams.MakeTestEncodingConfig()
+	std.RegisterLegacyAminoCodec(encodingConfig.Amino)
+	std.RegisterInterfaces(encodingConfig.InterfaceRegistry)
+	ModuleBasics.RegisterLegacyAminoCodec(encodingConfig.Amino)
+	ModuleBasics.RegisterInterfaces(encodingConfig.InterfaceRegistry)
+	return encodingConfig
+}
+
+// IBC Upgrade examples
+/*
 // setupUpgradeHandlers sets all necessary upgrade handlers for testing purposes
 func (app *SimApp) setupUpgradeHandlers() {
 	app.UpgradeKeeper.SetUpgradeHandler(
 		upgrades.V5,
-		upgrades.CreateDefaultUpgradeHandler(app.mm, app.configurator),
+		upgrades.CreateDefaultUpgradeHandler(app.ModuleManager, app.configurator),
```

Comment:
I think it would be good to keep the upgrade handlers uncommented. We make use of them in e2e tests for co-ordinated chain upgrades - see https://github.com/cosmos/ibc-go/blob/main/e2e/tests/upgrades/upgrade_test.go

All we need is that specific upgrade handlers are included in simapp on specific release versions, so all handlers are not strictly required (especially on main) but I think its no harm to register them on main rather than have them commented out.

Reply:
Hey, happy to work on getting these uncommented :)